lib/java/test-utils/src/main/java/org/enso/test/utils/MemoryUtils.java

```diff
@@ -11,18 +13,23 @@ final class MemoryUtils {
   private MemoryUtils() {}
 
   static void assertGC(String msg, boolean expectGC, Reference<?> ref) {
+    var buffer = new StringWriter();
+    var log = new PrintWriter(buffer);
     var memory = expectGC ? new ArrayList<>() : null;
     var retry = 3;
     for (var i = 1L; ; i *= 2) {
       try {
         var size = (int) Math.min(i, Integer.MAX_VALUE / 2);
         if (i >= 64) {
-          tryHarderToGc();
+          tryHarderToGc(log);
+          flushLog(buffer);
```

Comment:
- this is a _diagnostic_ `flushLog` 
- with it we can see in the logs that some tests are struggling to `assertGC`, but succeed at the end
- otherwise we'd only see the log on `OutOfMemoryError` and then the `.hprof` file is better source of info
- btw. read here on [instructions to analyze](https://github.com/enso-org/enso/pull/14219#discussion_r2476452008) the `.hprof` file

Reply:
- I see no traces of this `flushLog` in https://github.com/enso-org/enso/actions/runs/19422046657?pr=14301 run
- neither on Linux, nor Windows
- I don't see any Mac OS X runs however